File: eww/.config/eww/python-scripts/wifi.py
#!/usr/bin/python

import subprocess
from typing import Any


def print_flush(x: Any):
    print(x, flush=True)


def main():
    wifi_cmd = subprocess.run(
        ["nmcli", "-t", "-f", "active,ssid", "dev", "wifi", "list"],
        capture_output=True,
        text=True,
    )

    if wifi_cmd.returncode != 0:
        print_flush("unable to read wifi")

    search_for = "yes:"

    wifi_cmd_lines = wifi_cmd.stdout.split("\n")
    connected_ssids = list(
        map(
            lambda line: line.lstrip(search_for),
            filter(
                lambda line: line.startswith(search_for),
                wifi_cmd_lines,
            ),
        )
    )

    if len(connected_ssids) < 1:
        return print_flush("󰤯 No wifi?")

    ssid = connected_ssids[0]

    print_flush(f"󰤥 {ssid}")


if __name__ == "__main__":
    main()

File: eww/.config/eww/scripts/cmd/wifi/wifi.go
package wifi

import (
	"fmt"
	"os/exec"
	"strings"
)

func Get() {
	cmd := exec.Command("nmcli", "-t", "-f", "active,ssid", "dev", "wifi", "list")
	output, err := cmd.Output()

	if err != nil {
		fmt.Println("unable to read wifi")
		return
	}

	searchFor := "yes:"
	lines := strings.Split(string(output), "\n")

	var connectedSSIDs []string
	for _, line := range lines {
		if ssid, ok := strings.CutPrefix(line, searchFor); ok {
			connectedSSIDs = append(connectedSSIDs, ssid)
		}
	}

	if len(connectedSSIDs) < 1 {
		fmt.Println("󰤯 No wifi?")
		return
	}

	ssid := connectedSSIDs[0]
	fmt.Printf("󰤥 %s\n", ssid)
}
